doc/user/caching.xml

```diff
@@ -511,6 +511,48 @@ Program('prog',
 
   </section>
 
+  <section>
+    <title>Using a Custom CacheDir class</title>
+
+    <para>
+
+    SCons built-in &f-link-CacheDir; class can be extended to support customization
```

Comment:
the CacheDir this links to is the method, so it's not a class.  Don't think you can link to the class definition except in the API docs, so this should probably be written as `<classname>CacheDir</classname>` or `<classname>SCons.CacheDir.CacheDir</classname>`. Will leave to @bdbaddog whether something that's not technically public (at the moment) should be exposed in User Guide...

Reply:
Adding link to where this is in API docs just for reference: https://scons.org/doc/4.1.0/HTML/scons-api/SCons/?highlight=cachedir#SCons.CacheDir.CacheDir

Reply:
Hrmm.. do we reference any other such in the public docs?  
Generally we'd set such by strings which would get interpretted..

Reply:
And I won't even restart the discussion of how often SCons overloads things... it's a class, it's a method, it's a proxy, no, it's S(uper)Cons!

Reply:
made some small changes per @mwichmann recommendations in ffa9092